src/shared/components/case-editor/case-edit-submit/case-edit-submit.component.ts

```diff
@@ -109,30 +113,118 @@ export class CaseEditSubmitComponent implements OnInit, OnDestroy {
+            // such fields
+            if (caseField.value == null || !FieldsUtils.containsNonEmptyValues(caseField.value)) {
+              delete rawFormValueData[key];
+            }
+            break;
+          case 'Collection':
+            if (caseField.value != null) {
+              // If the collection is of a Complex field type, loop through each item and call this function recursively
+              // to check each Complex field instance's sub-fields, updating rawFormValueData for this instance
+              const collectionFieldType = caseField.field_type.collection_field_type;
+              if (collectionFieldType.type === 'Complex' && collectionFieldType.complex_fields.length > 0) {
+                if (caseField.retain_hidden_value) {
+                  (caseField.value as any[]).forEach((_, index) => {
+                    // Call this function recursively to check the Complex field's sub-fields
+                    const complexFormGroup = (formGroup.get(key) as FormArray).at(index).get('value') as FormGroup;
+                    const complexSubFields = caseField.field_type.collection_field_type.complex_fields;
+                    const resultantObject = this.filterRawFormValues(complexFormGroup, complexSubFields);
+                    // If the resultant object from the recursive function call is empty, remove the *existing item*
+                    // from the array in rawFormValueData altogether (effectively no update); else, update this item in
+                    // the array
+                    !FieldsUtils.isNonEmptyObject(resultantObject)
+                      ? rawFormValueData[key].splice(index, 1)
+                      : rawFormValueData[key][index].value = resultantObject;
+                  });
+
+                  // If the collection has been left empty as a result of removals above, remove the collection itself
+                  if (rawFormValueData[key].length == 0) {
+                    delete rawFormValueData[key];
+                  }
+                }
+              } else if (collectionFieldType.type !== 'Complex') {
+                if (caseField.retain_hidden_value) {
+                  // If retain_hidden_value = true on the collection then this applies to all items in the collection,
+                  // so remove the entire collection from rawFormValueData (effectively no update)
+                  delete rawFormValueData[key];
+                } else {
+                  // Otherwise, replace the collection with an empty array, which will clear the existing items
+                  rawFormValueData[key] = [];
+                }
+              }
+            }
+
+            // Discard null, undefined, or empty collections, to avoid submitting a superfluous update for such fields
+            if (caseField.value == null || !FieldsUtils.containsNonEmptyValues(caseField.value)) {
+              delete rawFormValueData[key];
+            }
+            break;
+          default:
```

Comment:
This _might_ need a specific case for `DynamicList`s simply because of the shenanigans with the `value` potentially being either an object or a string - if it's an object, it could still not have an _actual_ value defined. That's partly what @phillip-whitaker's recent change addresses.

Reply:
To be revisited under [EUI-3528](https://tools.hmcts.net/jira/browse/EUI-3528).